spec/Section 3 -- Type System.md

```diff
@@ -248,6 +248,29 @@ a given GraphQL server expects. Any other input value, including float input
 values (such as `4.0`), must raise a query error indicating an incorrect type.
 
 
+#### Any
+
+The Any scalar type represents any value that can be represented by underlying
+serialization protocol (including lists and maps). It is intended to be an opt-out type in case when the
+exact type of possible data is not known in advance or multiple types should be
+supported.
+
+Note: {null} value supplied as an argument or field value should be interpreted
+as literal {null} and handled according to GraphQL rules for arguments and
+field values.
+
+
+**Result Coercion**
+
+GraphQL servers should coerce raw value to the value that can be represented by
+underlying serialization protocol when possible otherwise they must raise a
+field error.
```

Comment:
@leebyron Idea was to specify that error should be raised in case if returned type can't be represented in serialization format, e.g. `NaN` in JS